internal/watcher/watcher.go

```diff
@@ -53,11 +57,19 @@ func (w *Watcher) Start(ctx context.Context, runner service.Runner) error {
 	decryptionDataChannel := make(chan *DecryptionKeysEvent)
 	keyShareChannel := make(chan *KeyShareEvent)
 
-	ethClient, err := ethclient.Dial(w.config.RpcURL)
+	dialer := rpc.WithWebsocketDialer(websocket.Dialer{
+		HandshakeTimeout: 45 * time.Second,
+		NetDial: (&net.Dialer{
+			Timeout:   45 * time.Second,
+			KeepAlive: 30 * time.Second,
```

Comment:
just used it for ensuring that idle connections are maintained rather than addressing issues with connection timing out. it has nothing to do with timeouts